relayer/src/light_client/tendermint.rs

```diff
@@ -56,6 +63,79 @@ impl super::LightClient<CosmosSdkChain> for LightClient {
 
         self.fetch_light_block(AtHeight::At(height))
     }
+
+    /// Given a client update event that includes the header used in a client update.
+    /// it looks for misbehaviour by fetching a header at same or latest height.
+    /// TODO - return also intermediate headers.
+    fn build_misbehaviour(
```

Comment:
Can we find a better name for this method and associated ones? To me the `build_` prefix implies that there is always a misbehavior to build whereas the actual behavior of the function is to figure out if there is a misbehavior by checking if the headers are compatible.

Perhaps

```suggestion
    fn check_misbehaviour(
```
?